lib/puppet/provider/mysql_user/mysql.rb

```diff
@@ -123,11 +125,7 @@ def password_hash=(string)
     else
       # Version >= 5.7.6 (many password related changes)
       if (mysqld_type == "mysql" or mysqld_type == "percona") and Puppet::Util::Package.versioncmp(mysqld_version, '5.7.6') >= 0
-        if string.match(/^\*/)
-          mysql([defaults_file, system_database, '-e', "ALTER USER #{merged_name} IDENTIFIED WITH mysql_native_password AS '#{string}'"].compact)
-        else
-          raise ArgumentError, "Only mysql_native_password (*ABCD...XXX) hashes are supported"
```

Comment:
According to the docs https://dev.mysql.com/doc/refman/5.7/en/create-user.html the forms are `IDENTIFIED WITH auth_plugin BY 'auth_string'` and `IDENTIFIED WITH auth_plugin AS 'hash_string'`

Your pull request only uses the `AS` keyword so I'm guessing is still hashed, but the example you gave in the description is not hashed. Should they be?

Would it make more sense to introduce an `auth_string` parameter for strings other than password hashes?